R/rdeephaven/src/client.cpp

```diff
@@ -96,6 +96,23 @@ class ClientOptionsWrapper {
         internal_options->setSessionType(sessionType);
     }
 
+    void setUseTls(bool useTls, std::string tlsRootCerts) {
+        internal_options->setUseTls(useTls);
+        internal_options->setTlsRootCerts(tlsRootCerts);
+    }
```

Comment:
Here, two methods have been combined into one.  This may be ok if we know that these two methods will always be used together, and there will be no other complexity added in the future.  If we expect things to possibly change in this area, some consideration should be made to breaking this up.  I don't know the tradeoffs and expectations on this.  

I guess, looking at this API, I might go with splitting this or setting useTls to true if the method is ever called.  It seems weird if you made the call `setUseTls(false, "ABCD")`.

Reply:
Cristian and I talked about this a bit, C++ does them separately because of how they chain the operations together in a call. My reasoning here is that `useTls` can be sensibly used without rootCerts, but `tlsRootCerts` will never (my understanding of Cristian's words) be used without a `useTls` to go along with it. In that case, I don't want an R user to be able to think about these as orthogonal methods, which I feel an API like `options$setUseTls` and `options$setTlsRootCerts` allows you to think. However, the API I have `options$useTls(optional rootCerts)`  communicates that you can use TLS without root certs, but there are no root certs without TLS.

In the R wrapping, I have this:
```
use_tls = function(root_certs = "") {
    self$internal_client_options$set_use_tls(TRUE, root_certs)
},
```

This would be just as good, and maybe better, as the C++ wrapping would mirror the C++ client:
```
use_tls = function(root_certs="") {
    self$internal_client_options$set_use_tls(TRUE)
    self$internal_client_options$set_tls_root_certs(root_certs)
}
```

Reply:
If this is all in the C++ wrapper interface, I would definitely go with the final option.  It is much more clear, and it reflects the C++ API better, which is what DH devs expect..